subsys/usb/class/dfu/usb_dfu.c

```diff
@@ -368,7 +393,11 @@ static void dfu_flash_write(uint8_t *data, size_t len)
 		LOG_DBG("flash write done");
 		dfu_data.state = dfuMANIFEST_SYNC;
 		dfu_reset_counters();
-		if (boot_request_upgrade(false)) {
+		/* TODO: Reexamine if there is a nicer way of handling mcuboot
+		 * interface
+		 */
+		if (dfu_data.fa->fa_id == FLASH_AREA_ID(image_1) &&
```

Comment:
What is user-case for this change? Currently boot_request_upgrade() is no-op if image_1 dosen't exist (so USB DFU is build in the MCUBoot for this case). It is the operation for dual-bank mode. Dual-XIP support is in progress - but mean also no-op for this function.